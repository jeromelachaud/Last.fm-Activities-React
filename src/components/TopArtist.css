/* TopArtist */
.TopArtist {
  margin-bottom: 5rem;
}

.TopArtist a:link,
.TopArtist a:visited {
  color: #111111;
  display: block;
  background-color: rgba(255, 255, 255, 0.8);
}

.TopArtist a:active {
  text-decoration: none;
}

.TopArtist a:hover {
  background-color: rgba(255, 255, 255, 1);
  -webkit-transition: background-color .3s ease-in-out;
  transition: background-color .3s ease-in-out;
}

.TopArtist__info {
  padding: 1rem 0 0.7rem 0.7rem;
  width: 300px;
}

.TopArtist__date {
  font-size: 80%;
}
