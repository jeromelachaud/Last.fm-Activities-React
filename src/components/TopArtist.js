import React from 'react';
import styled from 'styled-components';

const TopArtistWrapper = styled.ul`
  margin-bottom: 5rem;
    a:link,
    a:visited {
      color: #111111;
      display: block;
      background-color: rgba(255, 255, 255, 0.8);
    }

    a:active {
      text-decoration: none;
    }

    a:hover {
      background-color: rgba(255, 255, 255, 1);
      -webkit-transition: background-color .3s ease-in-out;
      transition: background-color .3s ease-in-out;
    }
`;

const TopArtistInfo = styled.ul`
  padding: 1rem 0 0.7rem 0.7rem;
  width: 300px;
`;

let TopArtist = React.createClass ({

  propTypes: {
    topArtist: React.PropTypes.object.isRequired
  },

  render() {
    const {
      topArtist
    } = this.props;

    const {
      name,
      playcount,
      image,
      url
    } = topArtist;

    return (
      <TopArtistWrapper>
        <li>
          <a href={url}>
            <img
              src={image['3']['#text']}
              alt={topArtist.name}/>
            <TopArtistInfo>
              <li><strong>{name}</strong></li>
              <li><i>{playcount} plays</i></li>
            </TopArtistInfo>
          </a>
        </li>
      </TopArtistWrapper>
    );
  }
});

export default TopArtist;
