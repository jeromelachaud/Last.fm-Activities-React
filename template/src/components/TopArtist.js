import React from 'react';
import './TopArtist.css';

let TopArtist = React.createClass ({

  render() {
    const {
      topArtist
    } = this.props;

    const {
      artist,
      image,
      url
    } = topArtist;

    return (
      <ul
        className="TopArtist">
        <li
          className="TopArtist_item">
          <a href={url}>
            <img src={image['3']['#text']} />
            <ul
              className="TopArtist__info">
              <li><strong>{artist['#text']}</strong></li>
              <li>{topArtist.name}</li>
            </ul>
          </a>
        </li>
      </ul>
    );
  }
});

export default TopArtist;
