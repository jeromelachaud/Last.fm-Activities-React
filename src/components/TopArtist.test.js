import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import TopArtist from './TopArtist';

const topArtist = {};

describe('<TopArtist> component', () => {
  const wrapper = shallow(<TopArtist topArtist={topArtist} />);

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
