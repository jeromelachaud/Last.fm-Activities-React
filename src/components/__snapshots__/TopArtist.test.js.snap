exports[`<TopArtist> component should render as expected 1`] = `
<styled.ul>
  <li>
    <a>
      <img
        src="" />
      <styled.ul>
        <li>
          <strong />
        </li>
        <li>
          <i>
             plays
          </i>
        </li>
      </styled.ul>
    </a>
  </li>
</styled.ul>
`;
